### PR TITLE
Get latest harvest records from db

### PR DESCRIPTION
**Pull request by rshewitt**
# Pull Request

Related to [#4744](https://github.com/GSA/data.gov/issues/4744)

## About

add latest harvest records read in db interface

## PR TASKS

- [x] The actual code changes.
- [x] Tests written and passed.
- [ ] Any changes to docs?
- [x] Bumped version number in [pyproject.toml](https://github.com/GSA/datagov-harvesting-logic/blob/8078c5b9f015ca7fb8a142e4e4b2e22ad2fd49b1/pyproject.toml#L3) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: tests/integration/database/test_db.py [hunk 1]
```diff
@@ -182,3 +190,26 @@ def test_filter_jobs_by_faceted_filter(
...
+        # `latest_records` fixture
+        interface.add_harvest_source(source_data_dcatus_2)
+        interface.add_harvest_job(job_data_dcatus)
+        interface.add_harvest_records(latest_records)
+
+        latest_records = interface.get_latest_records_by_source(
+            source_data_dcatus["id"]
+        )
+
+        assert len(latest_records) == 4
```
Comment (btylerburton) [inline tests/integration/database/test_db.py:277]: Would be nice to see more assertions to confirm we're getting the ones we want.

**Review by btylerburton — DISMISSED** [review]
Nice work!

File: tests/integration/database/test_db.py (hunk 1 above)
Comment (rshewitt) [reply tests/integration/database/test_db.py:277]: added

File: tests/conftest.py [hunk 2]
```diff
@@ -14,13 +14,15 @@
...
-HARVEST_SOURCE_URL = os.getenv('HARVEST_SOURCE_URL')
+HARVEST_SOURCE_URL = os.getenv("HARVEST_SOURCE_URL")
 
 
 @pytest.fixture(scope="session")
 def app() -> Flask:
     app = create_app()
 
+    app.config["TESTING"] = True
+
```
Comment (jbrown-xentity) [inline tests/conftest.py]: I don't think we need this anymore, do we?

**Review by jbrown-xentity — DISMISSED** [review]
Looks good, thanks!

File: tests/conftest.py (hunk 2 above)
Comment (rshewitt) [reply tests/conftest.py]: removed

**Review by jbrown-xentity — APPROVED** [review]

**Pull request merged**